### PR TITLE
Feature/350 implement showing multiple structures at the same time

### PR DESCRIPTION
**Pull request by Michel-Heinz**
I added a feature, where when a trajectory is loaded, all structures can be displayed at once. Therefore I enable loading multiple structures into the structure_widget. Please have a look and test a bit!

## Summary by CodeRabbit

- **New Features**
  - Enhanced the application to handle multiple structures simultaneously, improving user interaction with complex datasets.

- **Refactor**
  - Updated graphical rendering to support older OpenGL versions (3.3), ensuring broader compatibility with legacy systems.

- **Bug Fixes**
  - Adjusted test suites to align with the new multi-structure handling, ensuring reliability and correctness of functionality.

### CODE REVIEW

**Comment by Michel-Heinz** [comment]
What happened to code cov?

**Comment by ab5424** [comment]
> What happened to code cov?

It's there now. I think it can be delayed from time to time, depending on traffic.

**Review by ab5424 — APPROVED** [review]
Please open an issue describing the problems that might come up with the crystal functionality in the future due to these changes.

**Comment by Michel-Heinz** [comment]
I ll take a look at it. It sounds like a straight forward fix tho

**Comment by Michel-Heinz** [comment]
Is it possible that code cov does not see the pytest-qt tests? I cannot run them locally it tells me no tests were found.
Moreover, since I changed lines in untested code I'd have to write tests for these routines now. The problem is, I cannot get pytest-qt working.

**Comment by Michel-Heinz** [comment]
@adrianusler Any idea why tests on ubuntu are failing? I cannot figure it out

File: src/molara/Gui/structure_widget.py [hunk 1]
```diff
@@ -75,14 +75,18 @@
...
+            return True
+        return self.structures[0].draw_bonds and self.structures[0].has_bonds
 
     @property
     def draw_bonds(self) -> bool:
         """Specifies whether bonds should be drawn."""
-        return self.structure.draw_bonds
+        if len(self.structures) != 1:
+            return False
+        return self.structures[0].draw_bonds
```
[the remark this answers is not in the document]
Comment (adrianusler) [reply src/molara/Gui/structure_widget.py:89]: I am opening an issue for this so we don't forget about it.

File: src/molara/Gui/structure_widget.py [hunk 2]
```diff
@@ -75,14 +75,18 @@
...
     def bonds(self) -> bool:
         """Specifies whether bonds should be drawn (returns False if no bonds present whatsoever)."""
-        if self.structure_is_set:
-            return self.structure.draw_bonds and self.structure.has_bonds
-        return False
+        if not self.structures:
+            return False
+        if len(self.structures) > 1:
+            return True
+        return self.structures[0].draw_bonds and self.structures[0].has_bonds
```
[the remark this answers is not in the document]
Comment (adrianusler) [reply src/molara/Gui/structure_widget.py:82]: See issue #356.

File: src/molara/Gui/structure_widget.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (adrianusler) [reply src/molara/Gui/structure_widget.py:89]: See issue #356.

File: src/molara/Gui/structure_widget.py [hunk 3]
```diff
@@ -151,10 +159,11 @@
...
         """Centers the structure in the widget."""
-        if self.structure_is_set:
-            self.structure.center_coordinates()
-            self.camera.center_coordinates()
-            self.set_vertex_attribute_objects()
+        if not self.structures:
+            return
+        self.structures[0].center_coordinates()
+        self.camera.center_coordinates()
+        self.set_vertex_attribute_objects()
```
[the remark this answers is not in the document]
Comment (adrianusler) [reply src/molara/Gui/structure_widget.py:166]: It does handle this case (see lines 162 & 163).

File: src/molara/Gui/structure_widget.py [hunk 4]
```diff
@@ -354,11 +388,12 @@
...
-            self.structure.toggle_bonds()
-            self.set_vertex_attribute_objects()
-            self.update()
-            self.main_window.update_action_texts()
+        if len(self.structures) != 1:
+            return
+        self.structures[0].toggle_bonds()
+        self.set_vertex_attribute_objects()
+        self.update()
+        self.main_window.update_action_texts()
```
[the remark this answers is not in the document]
Comment (adrianusler) [reply src/molara/Gui/structure_widget.py:396]: See issue #356.

File: src/molara/Gui/structure_widget.py [hunk 5]
```diff
@@ -371,7 +406,7 @@
 
         :param update_box: specifies whether box shall be updated. If False, a drawn box will be hidden.
         """
-        if not self.structure_is_set:
+        if len(self.structures) != 1:
             return
 
         self.makeCurrent()
```
[the remark this answers is not in the document]
Comment (adrianusler) [reply src/molara/Gui/structure_widget.py:412]: We don't need to handle mixed cases with Crystal **and** Molecule objects. Right now, we don't have crystal trajectories yet, I think. We should look into that.

File: src/molara/Gui/structure_widget.py [hunk 6]
```diff
@@ -450,6 +485,9 @@
         :param xpos: x position of the mouse-click event
         :param ypos: y position of the mouse-click event
         """
+        if len(self.structures) != 1:
+            return -1
+
```
[the remark this answers is not in the document]
Comment (adrianusler) [reply src/molara/Gui/structure_widget.py:490]: Currently, we do not intend to implement the measurement features for cases with multiple structures being present.

**Review by adrianusler — APPROVED** [review]

**Pull request merged**